pkg/sensors/tracing/genericlsm.go

```diff
@@ -102,6 +113,72 @@ func (k *observerLsmSensor) LoadProbe(args sensors.LoadProbeArgs) error {
 		} else {
 			return err
 		}
+
+		if gl.imaHashProgLoad {
+			if gl.hook != "file_open" && gl.hook != "mmap_file" && gl.hook != "bprm_check_security" {
+				return fmt.Errorf("IMA hash collection for %s is not supported", gl.hook)
+			}
+			var btfSpec *btf.Spec
+			if btfFilePath := cachedbtf.GetCachedBTFFile(); btfFilePath != "/sys/kernel/btf/vmlinux" {
+				// Non-standard path to BTF, open it and provide it as 'KernelTypes'.
+				var err error
+				btfSpec, err = btf.LoadSpec(btfFilePath)
+				if err != nil {
+					return fmt.Errorf("opening BTF file '%s' failed: %w", btfFilePath, err)
+				}
+			}
+			spec, err := ebpf.LoadCollectionSpec(args.Load.Name)
```

Comment:
whatever solution we pick for this I think the IMA lsm hooks should go in separate object and we should be able to load it through normal LoadLSMProgram interface